xbmc/pvr/providers/PVRProvider.h

```diff
@@ -0,0 +1,232 @@
+/*
+ *  Copyright (C) 2012-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "addons/kodi-dev-kit/include/kodi/c-api/addon-instance/pvr/pvr_providers.h"
+#include "threads/CriticalSection.h"
+#include "utils/ISerializable.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace PVR
+{
+
+enum class ProviderUpdateMode
+{
+  BY_CLIENT,
+  BY_DATABASE
+};
+
+static constexpr int PVR_PROVIDER_ADDON_UID = -1;
+static constexpr int PVR_PROVIDER_INVALID_DB_ID = -1;
+
+class CPVRProvider final : public ISerializable
+{
+public:
+  CPVRProvider(int iUniqueId, int iClientId);
+  CPVRProvider(const PVR_PROVIDER& provider, int iClientId);
+  CPVRProvider(int iClientId,
+               const std::string& addonProviderName,
+               const std::string& addonIconPath,
+               const std::string& addonThumbPath);
+
+  bool operator==(const CPVRProvider& right) const;
+  bool operator!=(const CPVRProvider& right) const;
+
+  void Serialize(CVariant& value) const override;
+
+  /*!
+   * @brief The database id of this provider
+   *
+   * A unique identifier for this provider.
+   * It can be used to find the same provider on this clients backend
+   *
+   * @return The database id of this provider
+   */
+  int GetDatabaseId() const;
+
+  /*!
+   * @brief Set the database id of this provider
+   * @param iDatabaseId The new ID.
+   * @return True if the something changed, false otherwise.
+   */
+  bool SetDatabaseId(int iDatabaseId);
+
+  /*!
+   * @brief A unique identifier for this provider.
+   *
+   * A unique identifier for this provider.
+   * It can be used to find the same provider on this clients backend
+   *
+   * @return The Unique ID.
+   */
+  int GetUniqueId() const;
+
+  /*!
+   * @return The identifier of the client that supplies this provider.
+   */
+  int GetClientId() const;
+
+  /*!
+   * @return The name of the provider. Can be user provided or the backend name
+   */
+  std::string GetProviderName() const;
```

Comment:
Remove `Provider` from method name?! It is redundant, because the enclosing class is named `Provider`

Reply:
Done